src/whidbey/plugin.clj

```diff
@@ -1,29 +1,98 @@
 (ns whidbey.plugin
+  "This namespace runs inside of Leiningen and rewrites the project map to
+  include the customizations provided by Whidbey."
   (:require
     [leiningen.core.project :as project]))
 
 
-(defn whidbey-profile
-  [options]
-  `{:dependencies [[mvxcvi/puget "0.7.1"]
-                   [mvxcvi/whidbey "0.5.1"]]
+(defn- find-plugin-version
+  "Looks up the plugins in the project map and tries to find the version
+  specified for the given symbol. Returns nil if none matches."
+  [project plugin]
+  (try
+    (some (fn [[p v]] (when (= p plugin) v))
+          (:plugins project))
+    (catch Exception e
+      nil)))
 
-    :injections [(do (require 'whidbey.render)
-                     (alter-var-root
-                       #'whidbey.render/puget-options
-                       merge
-                       ~options))]
 
-    :repl-options {:nrepl-middleware [clojure.tools.nrepl.middleware.render-values/render-values]
-                   :nrepl-context {:interactive-eval {:renderer whidbey.render/render-str}}}})
+(defn- plugin-dependency
+  "Looks up the version for the given plugin, or sets it to `\"RELEASE\"`.
+  Returns a dependency vector entry."
+  [project plugin]
+  (vector
+   plugin
+   (or (find-plugin-version project plugin)
+       "RELEASE")))
+
+
+(defn- add-dependencies
+  "Adds some dependencies to the end of the current vector."
+  [project & deps]
+  (update-in project [:dependencies] concat deps))
+
+
+(defn- add-repl-init
+  "Adds the given repl initialization forms to the `:init` key in
+  `:repl-options`. Any existing initialization will happen before the
+  form is executed."
+  [project & forms]
+  (update-in project
+             [:repl-options :init]
+             (fn [current]
+               (if current
+                 `(do ~current ~@forms)
+                 `(do ~@forms)))))
+
+
+(defn- add-nrepl-middleware
```

Comment:
I'm not sure that you necessarily need to front-append given the changes to `clojure.tools.nrepl.middleware.render-values`.

...not that it makes much difference one way or the other - people tend not to have enough middleware for performance to matter here.